### PR TITLE
[add] community: add option to enable dynamic schema in Milvus vectorstores

### PR DESCRIPTION
**Pull request by rgupta2508**
- **Description:** add option to enable dynamic schema in milvus vectorstores, while creating collection  because in some cases we need to add some more field for existing collection. Currently Adding new field in not possible if collection is already created. 
        This will enable used to add dynamic schema field in case schema not finalised or field is not fixed for schema. 
    
    - **Issue:**  If field is not fixed for schema, it is difficult to add data in milvus. is we need to decide field dynamically based on running data.

- [x] **PR message**: ***Delete this entire checklist*** and replace with
    - **Description:** a description of the change
    - **Issue:** the issue # it fixes, if applicable
    - **Dependencies:** any dependencies required for this change
    - **Twitter handle:** if your PR gets announced, and you'd like a mention, we'll gladly shout you out!

An example notebook showing its use. It lives in `docs/docs/integrations` directory.

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

File: libs/community/langchain_community/vectorstores/milvus.py [hunk 1]
```diff
@@ -140,6 +142,7 @@ def __init__(
         replica_number: int = 1,
         timeout: Optional[float] = None,
         num_shards: Optional[int] = None,
+        enable_dynamic_field: Optional[bool] = False,
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/milvus.py:145]: can we just make this bool instead of Optional[bool]

**Review by baskaryan — COMMENTED** [review]
the langchain-community implementation of Milvus is deprecated in favor of the one in libs/partners/milvus, could we make any changes against that implementation

File: libs/community/langchain_community/vectorstores/milvus.py (hunk 1 above)
Comment (rgupta2508) [reply libs/community/langchain_community/vectorstores/milvus.py]: Yes, But i guess it will be a breaking changes for existing users.

Comment (rgupta2508) [reply libs/community/langchain_community/vectorstores/milvus.py]: Anyways community is deprecated, so user need to do change code to integrate to new langchain , so we can change .

**Comment by rgupta2508** [comment]
@efriis , I guess the error not related to this commit. not sure how to resolve.

Can you please review this, how can i fix this build error

**Comment by rgupta2508** [comment]
@efriis  Please review
numpy-1.24.* not supported in python 3.12 (https://pypi.org/project/numpy/1.24.0/)
 finally green signal after restricting version to 3.11
 cc @baskaryan

**Comment by rgupta2508** [comment]
@baskaryan please review, if this can be merged, or anything else we can do to support dynamic schema for langchain milvus object

**Comment by efriis** [comment]
Pinged it over to the milvus folks to get a look from their team!

**Comment by rgupta2508** [comment]
> Pinged it over to the milvus folks to get a look from their team!

Ok Thanks

**Comment by zc277584121** [comment]
@rgupta2508 After discussing with others in milvus team , we decide to adopt your idea of supporting dynamic schema for langchain_milvus, and deprecate metadata_field, because their functions are almost the same.  But dynamic schema is a more advanced concept in milvus, which we want to introduce to langchain_milvus. 
I will try to review this pr and compare the difference between dynamic schema and metadata_field in details, especially in practical usage cases. It will take few days. 
As for deprecating the arg `metadata_field`, @efriis , do you have any idea? I think it will involve backward compatibility issues. Are there any examples can refer to?

**Comment by rgupta2508** [comment]
> @rgupta2508 After discussing with others in milvus team , we decide to adopt your idea of supporting dynamic schema for langchain_milvus, and deprecate metadata_field, because their functions are almost the same. But dynamic schema is a more advanced concept in milvus, which we want to introduce to langchain_milvus. I will try to review this pr and compare the difference between dynamic schema and metadata_field in details, especially in practical usage cases. It will take few days. As for deprecating the arg `metadata_field`, @efriis , do you have any idea? I think it will involve backward compatibility issues. Are there any examples can refer to?

Ok Thanks

**Comment by efriis** [comment]
will close it for now and would encourage the milvus team to reopen from this branch with the changes they want!

**Comment by rgupta2508** [comment]
@efriis  Thanks, Do i need to tag anyone ?

**Comment by efriis** [comment]
it sounded like @zc277584121 and the milvus team would handle it!

**Comment by zc277584121** [comment]
Sorry for being late, I'm in the process of refactoring as previous decision, which will require some testing and debugging. it may be finished in few days.

**Comment by zc277584121** [comment]
hi @efriis , @rgupta2508 , I have finish it, please help to review it,  https://github.com/langchain-ai/langchain/pull/24544

**Comment by rgupta2508** [comment]
@zc277584121  Thanks

**Pull request closed without merging**